server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileResource.java

```diff
@@ -471,4 +473,12 @@ public void setProcessor(TsFileProcessor processor) {
   public TimeseriesMetadata getTimeSeriesMetadata() {
     return timeSeriesMetadata;
   }
+
+  public long getTimePartition() {
+    if (startTimeMap != null && !startTimeMap.isEmpty()) {
+      return StorageEngine.getTimePartition(startTimeMap.values().iterator().next());
+    }
+    String[] splits = FilePathUtils.splitTsFilePath(this);
+    return Long.parseLong(splits[splits.length - 2]);
```

Comment:
If we sync a tsfile from other iotdb, will this tsfile has time partition directory? Maybe we should use startTimeMap to get time partition

Reply:
Thanks, I use the startTimeMap first, then the path

Reply:
It is not guaranteed that the other IoTDB will use the same partition interval, so it is better to check the whole startTimeMap and endTimeMap to see if any device has crossed a partition and report an exception if so.

Reply:
restart and load could use different getPartition method, one check partition and the other not. I will add a method with partition check